ondemand_agentmanager.go

```diff
@@ -72,60 +77,84 @@ func (m *OnDemandAgentManager) Reconfigure(opts OnDemandAgentManagerReconfigureO
 
 // GetClusterAgent returns the Agent which is not bound to any bucket.
 func (m *OnDemandAgentManager) GetClusterAgent() (*Agent, error) {
-	m.lock.Lock()
-	defer m.lock.Unlock()
-
+	m.stateLock.Lock()
 	if m.closed {
+		defer m.stateLock.Unlock()
```

Comment:
nit: could move this to immediately after the lock to remove the duplication

Reply:
I agree.  It looks like this code wasn't meant to use the defer mechanism, but rather just unlock directly, but I think I went through and mostly changed our code to use deferral instead of manual unlocking except in cases where there are coordinated locks.

Reply:
Correct, copy and paste + lack of brain issue.